pybamm/solvers/processed_casadi_variable.py

```diff
@@ -0,0 +1,223 @@
+                    "Shape not recognized for {} ".format(base_variable)
+                    + "(note processing of 2D and 3D variables is not yet "
+                    + "implemented)"
+                )
+
+        # Make entries a function and compute jacobian
+        entries_MX = self.entries
+        self.casadi_entries_fn = casadi.Function(
+            "variable", [symbolic_inputs], [entries_MX]
+        )
+
+        # Don't compute jacobian if the entries are a DM (not symbolic)
+        if isinstance(entries_MX, casadi.DM):
+            self.casadi_sens_fn = None
+        # Do compute jacobian if the entries are symbolic (functions of input)
+        else:
+            sens_MX = casadi.jacobian(entries_MX, symbolic_inputs)
+            self.casadi_sens_fn = casadi.Function(
+                "variable", [symbolic_inputs], [sens_MX]
+            )
+
+    def initialise_0D(self):
+        "Create a 0D variable"
+        # Evaluate the base_variable index-by-index
+        for idx in range(len(self.t_sol)):
+            t = self.t_sol[idx]
+            u = self.u_sol[:, idx]
+            next_entries = self.base_variable(t, u, self.inputs)
+            if idx == 0:
+                entries = next_entries
+            else:
+                entries = casadi.horzcat(entries, next_entries)
+
+        self.entries = entries
+        self.dimensions = 0
+
+    def initialise_1D(self):
+        "Create a 1D variable"
+        len_space = self.base_eval.shape[0]
+        entries = np.empty((len_space, len(self.t_sol)))
+
+        # Evaluate the base_variable index-by-index
+        for idx in range(len(self.t_sol)):
+            t = self.t_sol[idx]
+            u = self.u_sol[:, idx]
+            next_entries = self.base_variable(t, u, self.inputs)
+            if idx == 0:
+                entries = next_entries
+            else:
+                entries = casadi.horzcat(entries, next_entries)
+
+        # Get node values
+        nodes = self.mesh[0].nodes
+
+        # assign attributes for reference (either x_sol or r_sol)
+        self.entries = entries
+        self.dimensions = 1
+        if self.domain[0] in ["negative particle", "positive particle"]:
+            self.first_dimension = "r"
+            self.r_sol = nodes
+        elif self.domain[0] in [
+            "negative electrode",
+            "separator",
+            "positive electrode",
+        ]:
+            self.first_dimension = "x"
+            self.x_sol = nodes
+        elif self.domain == ["current collector"]:
+            self.first_dimension = "z"
+            self.z_sol = nodes
+        else:
+            self.first_dimension = "x"
+            self.x_sol = nodes
+
+        self.first_dim_pts = nodes
+
+    def value(self, inputs=None, check_inputs=True):
+        """
+        Returns the value of the variable at the specified input values
+
+        Parameters
+        ----------
+        inputs : float, array-like, or dict
```

Comment:
can you provide more information about how input parameters are ordered in the input array? A dict does not have any particular order, so how does the input parameter keys get mapped to the vector?

Reply:
We could check `list(inputs.keys())` is the same as expected?